source/common/http/conn_manager_impl.h

```diff
@@ -561,7 +562,15 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     // Possibly increases buffer_limit_ to the value of limit.
     void setBufferLimit(uint32_t limit);
     // Set up the Encoder/Decoder filter chain.
-    bool createFilterChain();
+    enum class CreateFilterChainResult {
```

Comment:
This seems fairly complicated for the result we're aiming for.  Given this is only relevant for the HTTP/1 codec, what do you think about just removing upgrade: h2c there if present?

Reply:
+1 this sounds like a great idea. Thank you Alyssa!

/wait

Reply:
So, just leave the reset of the code the same and just check for then remove upgrade: h2c at the top of createFilterChain.   Ok.  This code does need some TLC since created_filter_chain_ can also mean that we are overloaded and the stat downstream_rq_ws_on_non_ws_route_ is incremented on overload as well.  However, I agree that we should not couple the cleanup and the issue fix.

Reply:
Done.  PTAL